tests/integration/kw_version_test.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+include './tests/unit/utils.sh'
+include './tests/integration/utils.sh'
+
+function oneTimeSetUp()
+{
+  setup_container_environment
+}
+
+function kw_version_test_helper()
+{
+  local distro="$1"
+  local container
+  local kw_git_dir
+  local kw_dir
+  local head_hash
+  local branch_name
+  local base_version
+  local expected_output
+  local output
+
+  kw_dir="${KWROOT_DIR}"
+  kw_git_dir="${kw_dir}/.git"
+
+  # In order to check correctness of `kw --version`, we collect some information
+  # from the git repo:
+  #
+  # - Base version (alpha, beta, or other)
+  # - Branch name
+  # - Commit sha
+  #
+  # Because the local KW repo is copied to the container, we run  the  following
+  # commands directly on the host instead of running in the container.
+  head_hash=$(git --git-dir "${kw_git_dir}" rev-parse --short HEAD)
+  branch_name=$(git --git-dir "${kw_git_dir}" rev-parse --short --abbrev-ref HEAD)
+  base_version=$(cat "${kw_dir}/src/VERSION" | head -n 1)
+
+  # using the gathered information, we build the expected output
+  expected_output=$(printf '%s\nBranch: %s\nCommit: %s' "$base_version" "$branch_name" "$head_hash")
+
+  # collect the kw version in the container
+  container="kw-${distro}"
+  output=$(container_exec "${container}" kw --version)
```

Comment:
To be safe, add single quotes around `'kw --version'`.

Reply:
It is not possible to add single-quotes around the kw commands because podman will try to run a file/binary/program called kw\ device\ --local instad of running kw and passing two arguments: device and --local.